### ISSUE TITLE
[NOT WORKING] "Bring Back the Borders!" mod and "Disable rounded corners in Windows 11" mod.

### ISSUE THREAD

**Issue by gaveitatry2** [ramensoftware/windhawk-mods#1077]
I saw someone's closed Issue on Feb 2024 claiming that he got it to work, but I am having no such luck.

I am using Windows 11 Pro 22H2 22621.4317 with Windhawk 1.5.1 (Portable version) that I got from the official Windhawk website.

I installed the mod. After installing the mod, I clicked on the Advanced tab for the mod and typed "dwm.exe" (without the quotation marks) in the "Custom process inclusion list" section and I clicked the Save button. I then restarted the computer and Windhawk several times. The other mods load, but not this one. I have Windhawk set to start with Windows by placing a Windhawk shortcut in the Startup folder.

I want to use the "Bring Back the Borders!" mod because it does both - disable rounded corners and brings back the borders. I only mentioned the other "Disable rounded corners in Windows 11" mod because it would be better than nothing to be able to disable rounded corners but not be able to do anything about the borders. I do not have both installed at the same time, so there is no chance of them conflicting with each other.

Mod author: @teknixstuff

**Comment by gaveitatry2** [comment]
Yeah, I saw that it had been updated, so I got my hopes up. But the update does not work. I also set the dxm.exe in the inclusion list. It would be great if @teknixstuff could fix it.

**Comment by teknixstuff** [comment]
> Same here, mod does not work at all. I have Windows 10 22H2.

Neither of those mods are applicable to Windows 10, as it already has the borders and corners.

**Comment by teknixstuff** [comment]
> I saw someone's closed Issue on Feb 2024 claiming that he got it to work, but I am having no such luck.
> 
> I am using Windows 11 Pro 22H2 22621.4317 with Windhawk 1.5.1 (Portable version) that I got from the official Windhawk website.

iirc, Windhawk portable cannot inject into system processes such as dwm.exe (although I might be wrong). It's worth trying with the installable version instead.

**Comment by gaveitatry2** [comment]
> iirc, Windhawk portable cannot inject into system processes such as dwm.exe (although I might be wrong). It's worth trying with the installable version instead.

@teknixstuff 

Thank you for your suggestion. The installable version is much better than the portable version. It makes the mods faster and more reliable. And your mod 'Bring Back the Borders!' seems to be working now because there are no more rounded borders. The borders are corners now. However, when it comes to the borders themselves, I was expecting the thick Windows 10 borders. Unfortunately, I only see the thin non-existence border that may be 1px (or could be 0px). I thought that the whole point of 'Bring Back the Borders!' was to bring back the borders that were removed in Windows 11. I did the dwm.exe stuff, and I restarted my computer. And I checked but the mod does not have anything listed under Settings. What am I missing?

**Comment by teknixstuff** [comment]
> > iirc, Windhawk portable cannot inject into system processes such as dwm.exe (although I might be wrong). It's worth trying with the installable version instead.
> 
> [@teknixstuff](https://github.com/teknixstuff)
> 
> Thank you for your suggestion. The installable version is much better than the portable version. It makes the mods faster and more reliable. And your mod 'Bring Back the Borders!' seems to be working now because there are no more rounded borders. The borders are corners now. However, when it comes to the borders themselves, I was expecting the thick Windows 10 borders. Unfortunately, I only see the thin non-existent borders that may be 1px (or could be 0px). I thought that the whole point of 'Bring Back the Borders!' was to bring back the borders that were removed in Windows 11. I did the dwm.exe stuff, and I restarted my computer. And I checked but the mod does not have anything listed under Settings. What am I missing?

It brings back the ability to have borders, but if your current msstyles do not contain any borders, it won't magically add them. You'll have to either edit your msstyles to add borders, or find another msstyles that does have them.

**Comment by gaveitatry2** [comment]
> It brings back the ability to have borders, but if your current msstyles do not contain any borders, it won't magically add them. You'll have to either edit your msstyles to add borders, or find another msstyles that does have them.

@teknixstuff 

After selected AeroLite in the Display settings, I was able to see the borders. Yay!

However, I ran into a problem. I have a lot of "resize window" macros for my browser, games, File Explorer, and other miscellaneous windows.

One example, is a macro that I created to resize my browser window to 1920x1033 an to position it at 0,0. But if I use your mod to enable borders and resize the browser window to 1920x1033, the borders won't show because they will be offscreen. To compensate, I have to change the macro to resize the browser window to 1906x1026 and offset it to position 7,0 (the borders seem to each be 7px thick). It would very inconvenient to manually change the numbers for every one of my macros because the new numbers are non intuitive. If I want to resize a window to 1000x500, I just want to use the numbers 1000 and 500 instead of tweaking them to something else.

According to a Grok AI, it thinks that:

"The mod likely uses DWM hooks (e.g., GetBorderRect) to define border thickness. It could be revised to offset the client area calculation, so a 1920px width includes borders (e.g., client area = 1906px, borders = 7px each side)."

What do you think. Is this something you can do? Adjust your mod so that the client are calculation includes the borders instead of excludes it?

Grok AI explanation here, for context:
https://grok.com/share/bGVnYWN5_29d43255-cfce-4d8e-8c23-c108e4a06b3a

**Comment by teknixstuff** [comment]
> According to a Grok AI, it thinks that:
> 
> "The mod likely uses DWM hooks (e.g., GetBorderRect) to define border thickness. It could be revised to offset the client area calculation, so a 1920px width includes borders (e.g., client area = 1906px, borders = 7px each side)."
> 
> What do you think. Is this something you can do? Adjust your mod so that the client area calculation includes the borders instead of excludes it?

Unfortunately not, or at least that is outside the scope of this mod. The mod just makes Windows use it's existing but disabled codepath for displaying windows with borders. The mod does not deal with any of the calculations for making that work, Windows's existing code does.

**Comment by gaveitatry2** [comment]
The macro program developer used Grok to share a PowerShell script with me that tells the macro program to make adjustments (make window 14px narrower, 7px shorter, and move the window to the right 7px). I just had to run the script once and now the windows are being positioned and sized right.

That plus you telling me that I needed the installation version instead of the portable version really helped. I thought that the mod was broken, but it was just a problem with the portable version. I noticed the other mods in the portable version are slow and unreliable but those mods worked perfectly on the installation version.

Thank you for your help. I am glad that I got this working.

**Comment by gaveitatry2** [comment]
@teknixstuff 

Just out of curiosity, is there a way to exclude adding borders to all programs except for a few that I add to the inclusion list? Or does it have to be every program on the computer? Reason I ask is because I like windows with borders when playing certain games. But for everything else, I am probably fine with just having W11 type borders. By the way, this is the first time that I am really using Windows 11. I had it installed on an HP EliteDesk mini computer, but I never used it. It was slow and laggy with very little memory and I did not have a mouse and keyboard hooked up to it right. But I finally updated my main computer to W11 and I'm trying out this border window stuff. I had to use that powershell script that you gave me to trick my macro program into considering the borders as part of the window instead of some extra addon attached to the outside of the window.
